newIDE/app/src/InstructionOrExpression/EnumerateInstructions.js

```diff
@@ -507,41 +509,32 @@ export const filterInstructionsList = <
   list: Array<T>,
   { searchText }: InstructionFilteringOptions
 ): Array<T> => {
-  if (!searchText === '') {
+  if (!searchText) {
     return list;
   }
 
-  const lowercaseSearch = searchText.toLowerCase();
-
-  const matchCritera = (enumeratedInstructionOrExpressionMetadata: T) => {
-    const {
-      displayedName,
-      fullGroupName,
-    } = enumeratedInstructionOrExpressionMetadata;
-    return (
-      displayedName.toLowerCase().indexOf(lowercaseSearch) !== -1 ||
-      fullGroupName.toLowerCase().indexOf(lowercaseSearch) !== -1
-    );
-  };
+  const directMatches = [];
+  const fuzzyMatches = [];
 
-  const favorExactMatch = (list: Array<T>): Array<T> => {
-    if (!searchText) {
-      return list;
+  list.forEach(option => {
+    // favor direct matches first
+    const lowercaseSearch = searchText.toLowerCase();
+    if (
+      option.displayedName.includes(searchText) ||
+      option.fullGroupName.includes(searchText)
```

Comment:
Previously we did the direct matches only against displayedName (not the group name). Might be worth keeping the previous behavior (a direct match of the group name is maybe not that useful) - and this make the search a tiny bit less complex.